Book-Your-Place/bookyourplace
# Move all guides routes to guides namespace

#### What's this PR do?
Moves all guides routes to guides namespace

##### Background context
Part of the work to secure all (non-public) routes.

This follows the other work that separated out the Guest-based routes.

Now controller in each of these namespaces can call the relevant Devise method: authenticate_guide! or authenticate_guest!

Follow up work: removes all the hard coded trip data; redirects guides to their trip#index view when signed in.


#### Where should the reviewer start?
(Sorry, this a large pull request, I found it hard to break it down into smaller PRs...) 

But the main area to focus on is:
app/controllers/guides/*

#### How should this be manually tested?
You should be able to visit your localhost root (eg: http://localhost:3000/guides) and sign in (will need to create a new guide in the console, or via: http://localhost:3000/guides/sign_up).

Then once signed in, will be able to visit: http://localhost:3000/guides/trips
to see all your trips.

#### Screenshots
![image](https://user-images.githubusercontent.com/1453680/51884690-401ded80-2380-11e9-8dfa-baeb15758f0a.png)


![image](https://user-images.githubusercontent.com/1453680/51884694-444a0b00-2380-11e9-8bc3-406e36eef289.png)

Brings up the test coverage to almost 100%:

![image](https://user-images.githubusercontent.com/1453680/51884797-b6baeb00-2380-11e9-9861-6e509d756183.png)

---

#### Migrations
None

#### Additional deployment instructions
Will need to restart server for new routes to take effect.

#### Additional ENV Vars
None
